python/python/tests/compat/venv_runner.py

```diff
@@ -0,0 +1,106 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright The Lance Authors
+
+"""
+Runner script executed inside virtual environments to run compatibility tests.
+
+This script runs as a persistent subprocess that accepts multiple method calls
+without restarting. This avoids the overhead of repeatedly importing Lance and
+its dependencies.
+
+Protocol:
+- Reads 4 bytes (message length as big-endian int)
+- Reads that many bytes (pickled tuple of (obj, method_name))
+- Executes method on object
+- Writes 4 bytes (response length)
+- Writes pickled response dict
+"""
+
+import os
+import pickle
+import struct
+import sys
+import time
+import traceback
+
+# Enable detailed timing output with DEBUG=1
+DEBUG = os.environ.get("DEBUG", "").lower() in ("1", "true", "yes")
+
+
+def read_message(stream):
+    """Read a length-prefixed pickled message from stream."""
+    # Read 4-byte length header
+    length_bytes = stream.buffer.read(4)
+    if len(length_bytes) < 4:
+        return None  # EOF
+
+    length = struct.unpack(">I", length_bytes)[0]
+
+    # Read message data
+    data = stream.buffer.read(length)
+    if len(data) < length:
+        raise RuntimeError(
+            f"Incomplete message: expected {length} bytes, got {len(data)}"
+        )
+
+    return pickle.loads(data)
+
+
+def write_message(stream, obj):
+    """Write a length-prefixed pickled message to stream."""
+    data = pickle.dumps(obj)
+    length = struct.pack(">I", len(data))
+    stream.buffer.write(length)
+    stream.buffer.write(data)
+    stream.buffer.flush()
+
+
+def main():
+    """Main loop that processes method calls until EOF."""
+    while True:
+        try:
+            # Read request (obj, method_name)
+            request = read_message(sys.stdin)
+            if request is None:
+                # EOF - parent closed connection
+                break
+
+            obj, method_name = request
+
+            # Execute method with timing
+            start_time = time.time()
+            if DEBUG:
+                print(
+                    f"[VENV TIMING] Executing {method_name}...",
+                    file=sys.stderr,
+                    flush=True,
+                )
+
+            method = getattr(obj, method_name)
```

Comment:
Why wrap the method in an object?  Why not just pickle the method itself?  Then you could use `__name__` to get the method name for logging.